components/shared/ui/menu.tsx

```diff
@@ -147,11 +147,17 @@ export const Menu: FC<Props> = (props) => {
   const [activeMenu, setActiveMenu] = useState<string | number>(-1);
   const [smallMenuActive, setSmallMenuActive] = useState(false);
   const [isScrolled, setIsScrolled] = useState(false);
+  const router = useRouter();
 
   const handleMenuClick = (menuId: string | number): void => {
     setActiveMenu(menuId === activeMenu ? -1 : menuId);
   };
 
+  useEffect(() => {
+    // set is scrolled to show the menu when scrolling is not possible
+    setIsScrolled(document.body.scrollHeight <= window.innerHeight);
+  }, [router]);

```

Comment:
Why do you need the router here? The component doesn't re-mount on navigating somewhere else? Also why do you use the whole `router` project as a dependency, isn't just the path property enough?